src/antelope/stores/evm.ts

```diff
@@ -102,7 +104,10 @@ export const useEVMStore = defineStore(store_name, {
                     provider.on('chainChanged', (newNetwork) => {
                         evm.trace('provider.chainChanged', newNetwork);
                     });
-                    provider.on('accountsChanged', (accounts) => {
+                    provider.on('accountsChanged', async (accounts) => {
+                        const network = useChainStore().currentChain.settings.getNetwork();
+                        await useAccountStore().loginEVM({ network });
+                        useBalancesStore().updateBalancesForAccount('logged', useAccountStore().loggedAccount);
                         evm.trace('provider.accountsChanged', accounts);
```

Comment:
After you log in the user, the balances and all data related to that account should be fetched automátically. That's the main idea, to delegate completely this task to the lib and forget about it.

```suggestion
                    provider.on('accountsChanged', async (accounts) => {
                        evm.trace('provider.accountsChanged', accounts);
                        const network = useChainStore().currentChain.settings.getNetwork();
                        useAccountStore().loginEVM({ network });
```